src/memberPublicKey.spec.ts

```diff
@@ -0,0 +1,211 @@
+        oid: TEST_OID,
+        publicKey: publicKey.toString('base64'),
+      });
+    });
+
+    test('Non existent id should return non existing error', async () => {
+      const invalidPublicKeyId = '111111111111111111111111';
+      await memberPublicKeyModel.create({
+        memberId: MEMBER_MONGO_ID,
+        oid: TEST_OID,
+        publicKey,
+      });
+
+      const result = await getMemberPublicKey(MEMBER_MONGO_ID, invalidPublicKeyId, serviceOptions);
+
+      requireFailureResult(result);
+      expect(result.reason).toBe(MemberPublicKeyProblemType.PUBLIC_KEY_NOT_FOUND);
+    });
+
+    test('Non existent member id should return non existing error', async () => {
+      const invalidMemberKeyId = '111111111111111111111111';
+      await memberPublicKeyModel.create({
+        memberId: MEMBER_MONGO_ID,
+        oid: TEST_OID,
+        publicKey,
+      });
+
+      const result = await getMemberPublicKey(invalidMemberKeyId, MEMBER_MONGO_ID, serviceOptions);
+
+      requireFailureResult(result);
+      expect(result.reason).toBe(MemberPublicKeyProblemType.PUBLIC_KEY_NOT_FOUND);
+    });
+
+    test('Record Find errors should be propagated', async () => {
+      const memberPublicKey = await memberPublicKeyModel.create({
+        memberId: MEMBER_MONGO_ID,
+        oid: TEST_OID,
+        publicKey: publicKey.toString('base64'),
+      });
+      await connection.close();
+
+      const error = await getPromiseRejection(
+        async () =>
+          getMemberPublicKey(MEMBER_MONGO_ID, memberPublicKey._id.toString(), serviceOptions),
+        Error,
+      );
+
+      expect(error).toHaveProperty('name', 'MongoNotConnectedError');
+    });
+  });
+
+  describe('deleteMemberPublicKey', () => {
+    test('Existing id should remove member public key', async () => {
+      const memberPublicKey = await memberPublicKeyModel.create({
+        memberId: MEMBER_MONGO_ID,
+        publicKey,
+        oid: TEST_OID,
+      });
+
+      const result = await deleteMemberPublicKey(memberPublicKey._id.toString(), serviceOptions);
+
+      requireSuccessfulResult(result);
+      const dbResult = await memberPublicKeyModel.findById(memberPublicKey._id);
+      expect(dbResult).toBeNull();
+      expect(mockLogging.logs).toContainEqual(
+        partialPinoLog('info', 'Member public key deleted', {
+          id: memberPublicKey.id,
+        }),
+      );
+    });
+
+    test('Non existing id should not remove any member public key', async () => {
+      const memberPublicKey = await memberPublicKeyModel.create({
+        memberId: MEMBER_MONGO_ID,
+        publicKey,
+        oid: TEST_OID,
+      });
+
+      const result = await deleteMemberPublicKey(PUBLIC_KEY_ID, serviceOptions);
+
+      requireSuccessfulResult(result);
+      const dbResult = await memberPublicKeyModel.findById(memberPublicKey.id);
+      expect(dbResult).not.toBeNull();
+    });
+
+    test('Record deletion errors should be propagated', async () => {
```

Comment:
We don't need this test here: We only need it in functions that could failed due to duplicated records (e.g., `createOrg()`).